src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -2167,11 +2289,21 @@ private void ImportFallthrough(BasicBlock next)
 
         }
 
-        private void CallRuntimeExport(TypeSystemContext context, string methodName)
+        private const string RuntimeExport = "RuntimeExports";
```

Comment:
Calling a RuntimeExport this way means we have to marshal arguments from the managed calling convention to native and back. Could we just set up a managed call to skip the extra marshaling?

Reply:
RuntimeExports are called through the normal convention unless you're calling it through an actual pInvoke definition. I agree that several of these routines could be short circuited using things like `GetNewObjectHelperForType` but at least it shouldn't be calling the generated reverse pInvokeable stub.